File: backend/app/schemas/recommendations.py
from pydantic import BaseModel, Field


class StarterPaperRecommendation(BaseModel):
    paper_id: str
    title: str
    authors: str
    year: int
    topics: list[str]
    difficulty: str
    score: float
    reason: str
    matched_interests: list[str] = Field(default_factory=list)
    matched_goals: list[str] = Field(default_factory=list)
    starter_question: str
    tags: list[str] = Field(default_factory=list)


class StarterRecommendationsResponse(BaseModel):
    recommendations: list[StarterPaperRecommendation]

File: frontend/src/services/recommendations.ts
import api from "./api";

export interface StarterPaperRecommendation {
  paper_id: string;
  title: string;
  authors: string;
  year: number;
  topics: string[];
  difficulty: "Foundational" | "Accessible" | "Intermediate";
  score: number;
  reason: string;
  matched_interests: string[];
  matched_goals: string[];
  starter_question: string;
  tags: string[];
}

export async function getStarterRecommendations(): Promise<StarterPaperRecommendation[]> {
  const response = await api.get<{
    recommendations: StarterPaperRecommendation[];
  }>("/recommendations/starter");

  return response.data.recommendations;
}
